src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/InstallPrePushHookTask.kt

```diff
@@ -0,0 +1,91 @@
+package org.jmailen.gradle.kotlinter.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import org.jmailen.gradle.kotlinter.support.findGitDir
+import org.jmailen.gradle.kotlinter.support.findGradlew
+import java.io.File
+
+/**
+ * Install or update the kotlinter-gradle pre-push hook.
+ */
+open class InstallPrePushHookTask : DefaultTask() {
+    @TaskAction
+    fun run() {
+        val dotGitDir = findGitDir(project.rootDir)
+        logger.info(".git directory: $dotGitDir")
+        val gradlew = findGradlew(project.rootDir).path
+
+        val hookDir = File(dotGitDir.absolutePath, "hooks").apply {
+            if (!exists()) {
+                logger.debug("Creating hook dir $this")
+                mkdir()
+            }
+        }
+        logger.info("hookDir: $hookDir")
+
+        val prePushHookFile = File(hookDir, "/pre-push").apply {
+            if (!exists()) {
+                logger.info("Creating $this anew")
+                createNewFile()
+                setExecutable(true)
+            }
+        }
+
+        if (prePushHookFile.length() == 0L) {
+            logger.info("Writing hook to empty file")
+            prePushHookFile.writeText(generateHook(gradlew, addShebang = true))
+        } else {
+            val prePushHookFileContent = prePushHookFile.readText()
+            val startIndex = prePushHookFileContent.indexOf(startHook)
+            if (startIndex == -1) {
+                logger.info("Appending hook to end of existing non-empty file")
+                prePushHookFile.appendText(generateHook(gradlew))
+            } else {
+                logger.info("Replacing existing hook")
```

Comment:
This is the case where we replace our own hook snippet? Is the use case I newer version of kotlinter-gradle with revised hook logic?

Reply:
Exactly.